src/main/java/org/broadinstitute/hellbender/tools/copynumber/utils/annotatedregion/SimpleAnnotatedGenomicRegionCollection.java

```diff
@@ -0,0 +1,165 @@
+package org.broadinstitute.hellbender.tools.copynumber.utils.annotatedregion;
+
+import com.google.common.collect.Lists;
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.broadinstitute.hdf5.Utils;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.copynumber.formats.collections.AbstractLocatableCollection;
+import org.broadinstitute.hellbender.tools.copynumber.formats.metadata.LocatableMetadata;
+import org.broadinstitute.hellbender.tools.copynumber.formats.metadata.SimpleLocatableMetadata;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableReader;
+import org.broadinstitute.hellbender.utils.tsv.TableUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.broadinstitute.hellbender.tools.copynumber.utils.annotatedregion.SimpleAnnotatedGenomicRegion.*;
+
+/**
+ * Represents a collection of annotated regions.  The annotations do not need to be known ahead of time, if reading from a file.
+ * Though {@link SimpleAnnotatedGenomicRegion::CONTIG_HEADER},
+ *  {@link SimpleAnnotatedGenomicRegion::START_HEADER}, and
+ *  {@link SimpleAnnotatedGenomicRegion::END_HEADER} are always expected for defining the genomic region.
+ */
+public class SimpleAnnotatedGenomicRegionCollection extends AbstractLocatableCollection<LocatableMetadata, SimpleAnnotatedGenomicRegion> {
```

Comment:
This class looks very different from the other CNV collections classes.  All other classes are essentially an enum of column headers, codecs specified within the class (either statically or dynamically), and constructors that take either a file or metadata + list and simply call the superclass's constructor.  They contain no custom read/write methods, as these should all be handled by the superclass.

In contrast, your constructors include mandatory columns and codecs and you introduce several custom read/write methods.

I also think the use of Javadoc actually makes the class harder to read (and maintain).  These classes should be simple and self documenting.

Reply:
Let me look into AnnotatedInterval.  Then I might be able to get what I need there.